tests/test_dataclasses.py

```diff
@@ -1434,7 +1434,7 @@ def double_b(cls, v, _):
     assert d.b == 5.0
 
 
-@pytest.mark.xfail(reason='working on V2 - validator in child not applied')
+@pytest.mark.xfail(reason='validator in child not applied')
```

Comment:
If this means that `B`'s `validate_a` method is not called it sounds like a bug in the collection of validators and should be an easy fix.